integration/tests/cook/test_basic.py

```diff
@@ -625,27 +612,23 @@ def test_multiple_progress_updates_submit(self):
                   f'{line_3} && sleep 2 && {line_4} && sleep 2 && ' \
                   f'{line_5} && sleep 2 && echo "Done" && sleep 10 && exit 0'
         job_uuid, resp = util.submit_job(self.cook_url, command=command, executor=job_executor_type,
-                                         max_runtime=60000, max_retries=5)
+                                         max_runtime=60000, max_retries=5,
+                                         progress_output_file='progress_file.txt')
         self.assertEqual(201, resp.status_code, msg=resp.content)
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         message = json.dumps(job, sort_keys=True)
         self.assertEqual('success', job['state'], message)
-
-        instance = util.wait_for_sandbox_directory(self.cook_url, job_uuid, 'success')
-        message = json.dumps(instance, sort_keys=True)
-        self.assertIsNotNone(instance['output_url'], message)
-        self.assertIsNotNone(instance['sandbox_directory'], message)
-        self.assertEqual('cook', instance['executor'])
-        util.sleep_for_publish_interval(self.cook_url)
```

Comment:
Is the exit code going to be populated after all the progress reporting has completed? If not, we might still need the sleep.

Reply:
I don't understand what `sleep_for_publish_interval` is doing here.

We sleep/retry until the exit code shows up. That part's unchanged. Now we _also_ sleep/retry until the progress percentage we want shows up—which I thought would be redundant with this call to `sleep_for_publish_interval`.

Do we need both?

Reply:
We discussed this offline and we both concluded that the extra sleep here is not necessary.

However, I did add a new `time.sleep` call to a couple of the tests. Since the jobs they run include sleep commands too, it seems like the test should sleep for the same amount of time before it starts polling for a successful instance.